traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
+    """ Event object to represent mutations to a list.
+
+    Attributes
+    ----------
+    trait_list : list
+        The list being mutated.
```

Comment:
Whether the list contains content before or after the change depends on whether the change event is fired before or after the mutation, because it is the same list object. Strictly speaking, the event object does not need to dictate when the notification should be fired. Not sure how I could write this...

Reply:
We could add to the API document (not here) that notifications are fired after the mutations. What do you think?
i.e. `TraitList.notify` are called after the mutation has taken place.

Reply:
Oh, okay, I thought the behaviour was consistent. I think it would be good to report which events are fired before changes and which are fired after, but I agree that the event itself is not a place for that.

Reply:
I am sorry I did not mean to imply there were events fired before changes. All events are fired after changes.  Sorry for the confusion there.

Reply:
Then maybe having past tense tense in all 4 attribute descriptions would be a good way to say that the changes have happened without "dictating when notification is fired"?

Reply:
I was meaning to say that the event object is agnostic about whether the change has happened or is about to happen. The "added" and "removed" descriptions are intentionally ambiguous.  We should be able to reuse this object to represent a change that is about to happen (that is not a use case, but there is no need for this event to prevent that).

Reply:
Okay, well I would still argue that it would be good to inform the user that the event is fired after changes are made, but then I really don't know what would be the right place to do so.

Reply:
Yes I agree that the information needs to be somewhere in the documentation. In fact, I would very much like that to be included in the user manual.

While it may be convenient to have that description in the event object parameter section, doing so may introduce restrictions that do not belong to the data structure.

Reply:
Actually, the first sentence in the current user manual already implies that all change notifications are fired after the change:

https://docs.enthought.com/traits/traits_user_manual/notification.html

Reply:
I see that similar implicit mentions are made in the new observer's user manual as well, so it might be enough. I was just a bit worried that because the handlers are now supposed to deal with the actual events, the user's might be paying more attention to the docstrings of those events.

Reply:
It might not be amplified enough, but it is a fairly strong feature of Traits that notifications _always_ fire after the value has set.